src/core/agent_space_interaction.jl

```diff
@@ -185,3 +185,49 @@ function add_agent_single!(model::ABM, properties...)
     return agent
   end
 end
+
+
+####################################
+### sample
+####################################
+
+"""
+        sample!(model::ABM [, weight]; kwargs...)
+
+Replace agents of the `model` with a random sample of the current agents with
+size `n`. Optionally, choose an agent property `weight`(Symbol) to weight the sampling.
+
+# Keywords
+
+* `n`: the number of new agents.
+* `replace=true` whether sampling is performed with replacement.
+* `rng`::AbstractRNG=Random.GLOBAL_RNG: Optionally, specify a random number generator.
+"""
+function sample!(model::ABM, weight=nothing; n::Int, replace=true, ordered=false, rng::AbstractRNG=Random.GLOBAL_RNG)
```

Comment:
Keyword `ordered` exists in the function but is not documented. What does it achieve?

Reply:
It should be removed. I overlooked it.